### PR TITLE
no need to get hostname and nodename in func 'NewMainKubelet'

### PR DESCRIPTION
**Pull request by Pingan2017**
**What this PR does / why we need it**:
no need to get hostname and node in func 'NewMainKubelet' , we get them in the previous func
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -1061,7 +1063,9 @@ func CreateAndInitKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 	nodeLabels map[string]string,
 	seccompProfileRoot string,
 	bootstrapCheckpointPath string,
-	nodeStatusMaxImages int32) (k kubelet.Bootstrap, err error) {
+	nodeStatusMaxImages int32,
+	hostname string,
```
Comment (liggitt) [inline cmd/kubelet/app/server.go]: adding hostname when we already have a hostnameOverride parameter is a little crazy-making

Comment (Pingan2017) [reply cmd/kubelet/app/server.go]: Yeah , but only paasing nodename is not clean. 
i will close this pr, thanks ! @liggitt

**Pull request closed without merging**